core/src/api/api.rs

```diff
@@ -417,11 +417,20 @@ impl GraphState {
 		}
 		// import bundles
 		for ImportBundle { schema_id, pages, dsnp_keys, dsnp_user_id, key_pairs } in payloads {
-			let connection_type = self
-				.environment
-				.get_config()
-				.get_connection_type_from_schema_id(*schema_id)
-				.ok_or(DsnpGraphError::InvalidSchemaId(*schema_id))?;
+			let connection_type_option =
+				self.environment.get_config().get_connection_type_from_schema_id(*schema_id);
+			let connection_type = match connection_type_option {
+				Some(connection_type) => connection_type,
+				None => {
+					if pages.is_empty() && !key_pairs.is_empty() && !dsnp_keys.is_none() {
+						// This condition implies that only keys are being imported
+						ConnectionType::Follow(PrivacyType::Private)
+					} else {
+						return Err(DsnpGraphError::InvalidSchemaId(*schema_id))
+					}
+				},
+			};
```

Comment:
Here the very specific condition is checked
if we dont have schemaId and pages are empty then if and only if `dsnpKeys` and `keyPairs` are provided, we proceed else throw the error

Reply:
In this part, its more like a safe bypass for only importing the keys